src/vanilla/ClientModelExtensions.cs

```diff
@@ -970,5 +970,148 @@ public static bool ContainsTimeSpan(this CompositeType type)
 
             return prop != null;
         }
+
+        public static string InitializeType(this IModelType paramType, string paramName, bool isBrowser = false)
+        {
+            if (paramType is EnumType)
+            {
+                return paramType.InitializeEnumType(isBrowser);
+            }
+            else if (paramType is PrimaryType)
+            {
+                return paramType.InitializePrimaryType(paramName, isBrowser);
+            }
+            else if (paramType is SequenceType)
+            {
+                return paramType.InitializeSequenceType(paramName, isBrowser);
+            }
+            else if (paramType is DictionaryType)
+            {
+                return paramType.InitializeDictionaryType(paramName, isBrowser);
+            }
+            else if (paramType is CompositeType)
+            {
+                return paramType.InitializeCompositeType(paramName, isBrowser);
+            }
+
+            return null;
+        }
+
+        public static string InitializeCompositeType(this IModelType paramType, string paramName, bool isBrowser = false)
+        {
+            var builder = new IndentedStringBuilder("  ");
+            var composite = (CompositeType)paramType;
+            if (composite.ComposedProperties.Any())
+            {
+                builder.AppendLine("{").Indent();
+                var composedPropertyList = new List<Property>(composite.ComposedProperties.Where(p => /**p.IsRequired &&**/ !p.IsReadOnly));
+                for (var i = 0; i < composedPropertyList.Count; i++)
+                {
+                    var prop = composedPropertyList[i];
+                    if (i != composedPropertyList.Count - 1)
+                    {
+                        builder.AppendLine("{0}: {1},", prop.Name, prop.ModelType.InitializeType(prop.Name, isBrowser));
+                    }
+                    else
+                    {
+                        builder.AppendLine("{0}: {1}", prop.Name, prop.ModelType.InitializeType(prop.Name, isBrowser));
+                    }
+                }
+                builder.Outdent().Append("}");
+            }
+            else
+            {
+                builder.AppendLine("{}");
+            }
+            return builder.ToString();
+        }
+
+        public static string InitializeDictionaryType(this IModelType paramType, string paramName, bool isBrowser = false)
+        {
+            var dictionary = (DictionaryType)paramType;
+            var paramValue = $"{{ \"key1\": {dictionary.ValueType.InitializeType(paramName, isBrowser)} }}";
+            return paramValue;
+        }
+
+        public static string InitializeSequenceType(this IModelType paramType, string paramName, bool isBrowser = false)

```

Comment:
What does initializing the type mean?

Reply:
Is this about providing default values for all the parameters to put in the generated sample?

Reply:
It means that code will be generated/return for initializing a variable with that AutoRest type.